test/Microsoft.AspNetCore.Razor.Runtime.Test/Runtime/TagHelpers/TagHelperRunnerTest.cs

```diff
@@ -216,10 +216,9 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
 
                 output.TagName = "foo";
 
-                TagHelperAttribute classAttribute;
-                if (output.Attributes.TryGetAttribute("class", out classAttribute))
+                if (output.Attributes.ContainsName("class"))
                 {
-                    classAttribute.Value = "somethingelse";
+                    output.Attributes.SetAttribute("class", "somethingelse");
```

Comment:
Hmm, we might need a `ReadOnlyTagHelperAttributeList.IndexOf(string)` method.

The old code only changed the first attribute and did not remove additional matches. More like:
``` c#
    var index = output.Attributes.IndexOf("class");
    if (index >= 0)
    {
        output.Attributes[index] = new TagHelperAttribute("class", "something else");
    }
```